### DISCUSSION TITLE
Types generation from smithy models

### DISCUSSION THREAD

**Discussion by jpbourgeon** [alchemy-run/distilled#29] [Ideas]
As stated in their [README](https://www.npmjs.com/package/aws-sdk), AWS SDK v2 is going into maintenance mode in 2023. At some point in the future, type generation from its API may become incomplete or even unreliable. On the other side we already use smithy models in the project.

Would it be possible to generate type declarations off of the smithy models instead of AWS SDK v2? 

To go further, since [Smithy](https://smithy.io/2.0/index.html), is made for generating code, could we write our own community driven smithy models for the missing services we need (such as S3, SNS, and SQS) instead of handling them in the core SDK code? This would keep the core SDK functionality ubiquitous and maintainable.

**Comment by sam-goodwin** [comment]
I think the most important thing for us to look into is handling each of the protocols. Most are basic json which can he handled generically with proxies but some are using very strange protocols, like S3 and SQS. We should start to enumerate them, implement a core protocol module for each. Then we can generate a comprehensive SDK from smithy.
